alf/optimizers/optimizers.py

```diff
@@ -92,8 +166,99 @@ def step(self, closure=None):
             else:
                 tensor_utils.clip_by_norms(
                     grads, self._gradient_clipping, in_place=True)
+
+        if self._parvi is not None:
```

Comment:
Use a separate function to do parvi so the code is cleaner.

Reply:
Updated.